### PR TITLE
Fix validate contents on delete

### PR DESCRIPTION
**Pull request by pschoffer**
## Description
Validate contents does not heal correctly documents missing in index, because it would be marked as inconsistent record and then indexed instead of deleted.

### Plugin Updated: Enterprise Search

Updates `wp vip-search validate-contents` to correctly delete posts that are extra in ElasticSearch index.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test

1) Create dev-env - `vip dev-env create --slug test --mu-plugins /home/pavel/git/automattic/vip-go-mu-plugins`
1) Start dev-env - `vip dev-env start --slug test`
1) Enable search - `vi /home/pavel/.local/share/vip/dev-environment/test/config/wp-config-defaults.php` && uncomment search constants
1) Index the site - `vip dev-env --slug test exec -- wp vip-search index --setup`
1) Create a post - `vip dev-env --slug test exec -- wp post create --post_title=test --post_status=publish`
2) Delete a post directly in DB - `vip dev-env --slug test exec -- wp db cli` + `DELETE FROM wp_posts WHERE ID = 9;`
3) Flush the cache - `vip dev-env --slug test exec -- wp cache flush`
4) Run `dev-env exec --slug test -- wp vip-search health validate-contents --last_post_id=100` (we are doing last_post_id becouse otherwise it would take the last ID [from DB ](https://github.com/Automattic/vip-go-mu-plugins/blob/467244671acdd70470c3c397085a2fcd677cd535/search/includes/classes/class-health.php#L482) which is too low after we deleted the last one )
5) Run `dev-env exec --slug test -- wp vip-search health validate-contents --last_post_id=100` again this time all is well.

### CODE REVIEW

**Comment by rebeccahum** [comment]
@pschoffer Unable to reproduce the issue right now. Using just master, I'm seeing that it corrected it. I deleted post ID 21 and on the second `validate-contents` run it was gone:

```
╭─rebasaurus@rebatr0n32 ~/vipdev/single-site/vip-go-mu-plugins ‹master›
╰─$ vip dev-env --slug single-site exec --  wp vip-search health validate-contents
Validating posts 1 - 47...✘ (attempted to reconcile)
Warning: Inconsistencies fixed:
type,id,issue
post,16,missing_from_index
post,21,extra_in_index
╭─rebasaurus@rebatr0n32 ~/vipdev/single-site/vip-go-mu-plugins ‹master›
╰─$ vip dev-env --slug single-site exec --  wp vip-search health validate-contents
Validating posts 1 - 47...✘ (attempted to reconcile)
Warning: Inconsistencies fixed:
type,id,issue
post,16,missing_from_index
```

```
╭─rebasaurus@rebatr0n32 ~/vipdev/single-site/vip-go-mu-plugins ‹master›
╰─$ vip dev-env --slug single-site exec --  wp vip-search documents get post 21                                                                            130 ↵
Error: Document with ID 21 of type post was not found.
Error: Error: Document with ID 21 of type post was not found.
```

**Comment by pschoffer** [comment]
Hmm weird... I tried again and managed to reproduce again. I created more detailed steps to test. Please let me know if that worked.

I also notice that my fix didn't really work :) So I fixed that.

File: search/includes/classes/class-health.php [hunk 1]
```diff
@@ -632,7 +637,7 @@ public function validate_index_posts_content_batch( $indexable, $start_post_id,
 			                : self::simplified_diff_document_and_prepared_document( $document, $prepared_document ); // phpcs:ignore Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed
 
 			if ( $diff ) {
-				$key           = self::get_post_key( $document['ID'] );
+				$key = self::get_post_key( $document['ID'] );
 				$diffs[ $key ] = $inspect ? $diff : self::simplified_format_post_diff( $document['ID'], 'inconsistent' );
```
Comment (sjinks) [inline search/includes/classes/class-health.php:641]: This will trigger a PHPCS warning.

**Comment by rebeccahum** [comment]
> (we are doing last_post_id becouse otherwise it would take the last ID from DB which is too low after we deleted the last one )

I'm wondering if we should check if the last ID from both DB and ES use whichever is higher instead.

**Review by rebeccahum — APPROVED** [review]
Tested and works great!

File: search/includes/classes/class-health.php (hunk 1 above)
Comment (rebeccahum) [reply search/includes/classes/class-health.php]: I added a commit to fix it.

**Comment by rebeccahum** [comment]
r2028-stacks

**Pull request merged**